packages/core/src/codewhisperer/service/transformByQ/humanInTheLoopManager.ts

```diff
@@ -0,0 +1,102 @@
+    private readonly tmpDownloadsDir = path.join(this.osTmpDir, this.tmpDownloadsFolderName)
+
+    private tmpSessionFiles: string[] = []
+    private pomFileVirtualFileReference!: vscode.Uri
+    private manifestFileValues!: IManifestFile
+    private newPomFileVirtualFileReference!: vscode.Uri
+
+    public getTmpDependencyListDir = () => this.tmpDependencyListDir
+    public getUserDependencyUpdateDir = () => this.userDependencyUpdateDir
+    public getTmpDownloadsDir = () => this.tmpDownloadsDir
+    public getPomFileVirtualFileReference = () => this.pomFileVirtualFileReference
+    public getManifestFileValues = () => this.manifestFileValues
+    public getNewPomFileVirtualFileReference = () => this.newPomFileVirtualFileReference
+
+    public setPomFileVirtualFileReference = (file: vscode.Uri) => (this.pomFileVirtualFileReference = file)
+    public setManifestFileValues = (manifestFileValues: IManifestFile) => (this.manifestFileValues = manifestFileValues)
+    public setNewPomFileVirtualFileReference = (file: vscode.Uri) => (this.newPomFileVirtualFileReference = file)
+
+    public getUploadFolderInfo = (): FolderInfo => {
+        return {
+            name: this.userDependencyUpdateFolderName,
+            path: this.userDependencyUpdateDir,
+        }
+    }
+
+    public getCompileDependencyListFolderInfo = (): FolderInfo => {
+        return {
+            name: this.tmpDependencyListFolderName,
+            path: this.tmpDependencyListDir,
+        }
+    }
+
+    public getDependencyListXmlOutput = async () =>
+        await fsCommon.readFileAsString(path.join(this.tmpDependencyListDir, this.localPathToXmlDependencyList))
+
+    public createPomFileCopy = async (outputDirectoryPath: string, pomFileVirtualFileReference: vscode.Uri) => {
+        const newPomCopyRef = await createPomCopy(outputDirectoryPath, pomFileVirtualFileReference, 'pom.xml')
+        this.tmpSessionFiles.push(newPomCopyRef.path)
+        return newPomCopyRef
+    }
+
+    public replacePomFileVersion = async (pomFileVirtualFileReference: vscode.Uri, version: string) =>
+        await replacePomVersion(pomFileVirtualFileReference, version, this.pomReplacementDelimiter)
+
+    public cleanUpArtifacts = async () => {
+        const artifactCleanUpErrorMessage = (e: any) =>
+            `CodeTransformation: Error cleaning up artifacts = ${e?.message}`
+        try {
+            await fsCommon.delete(this.userDependencyUpdateDir)
+        } catch (e: any) {
+            getLogger().error(artifactCleanUpErrorMessage(e))
+        }
+        try {
+            await fsCommon.delete(this.tmpDependencyListDir)
+        } catch (e: any) {
+            getLogger().error(artifactCleanUpErrorMessage(e))
+        }
+        try {
+            await fsCommon.delete(this.tmpDownloadsDir)
+        } catch (e: any) {
+            getLogger().error(artifactCleanUpErrorMessage(e))
+        }
```

Comment:
this could be a single try/catch

Reply:
`fsCommon.delete` throws an error no? We want the minimal impact for artifact deletion if something went wrong.